value.go

```diff
@@ -725,6 +726,10 @@ func (vlog *valueLog) replayLog(lf *logFile, offset uint32, replayFn logEntry) e
 	if err != nil {
 		return errFile(err, lf.path, "Unable to replay logfile")
 	}
+	// The entire file should be truncated (i.e. it should be deleted)
+	if endOffset == 0 {
+		return errDeleteVlogFile
```

Comment:
Once this error is returned, we delete the vlog file. We should return this error once we check DB was opened with vlog.opt.Truncate true.

Reply:
Yeah. Makes sense. Fixed in https://github.com/dgraph-io/badger/pull/823/commits/7104d7d8aedf3eee6c3a07c5e7ca841e8c6bfa1d